expo/expo
# [docs] Clean up navigation constant to render new sidebar navigation

# Why

Part of [ENG-3962](https://linear.app/expo/issue/ENG-3962/add-new-components-as-separate-prs-in-expoexpo), follow-up PR #16192

This cleans up the navigation constants to make the new sidebar render properly.

# How

Previously, we had [custom code to detect if we should skip group titles](https://github.com/expo/expo/blob/sdk-44/docs/components/DocumentationSidebar.tsx#L33-L49). With the latest refactor to unist tree format, and dynamic renderers in, we can just skip adding these groups if we don't want to render them.

They are all pretty much interchangeable, you can render a section, group, or page.

# Test Plan

- Open random pages
- Check if sidebar is rendered properly
  - No section/group titles that are equal

# Checklist
